content/javascript/concepts/arrays/terms/copyWithin/copyWithin.md

```diff
@@ -0,0 +1,87 @@
+---
+Title: '.copyWithin()'
+Description: 'Copies part of an array to another location in the same array and returns it without modifying its length.'
+Subjects:
+  - 'Web Development'
+  - 'Computer Science'
+Tags:
+  - 'Arrays'
+  - 'Methods'
+CatalogContent:
+  - 'introduction-to-javascript'
+  - 'paths/front-end-engineer-career-path'
+---
+
+The `.copyWithin()` copies part of an array to another location in the same array and returns it without modifying its length. The `.copyWithin()` method is a mutating method. It does not alter the length of the array, but it will change the content of the array and create new properties or delete existing properties, if necessary.
+
+
+## Syntax
+
+```pseudo
+array1.copyWithin(target);
+array1.copyWithin(target, start);
+array1.copyWithin(target, start, end);
+```
+
+### Parameters
+
+- `target`: Zero-based index at which to copy the sequence to. 
+  - Negative index counts back from the end of the array — if `target < 0`, `target + array.length` is used.
+  - If `target < -array.length`, `0` is used.
+  - If `target >= array.length`, nothing is copied. 
+  - If `target` is positioned after `start` after normalization, copying only happens until the end of `array.length`(in other words, `copyWithin()` never extends the array).
+
+- `start` (optional): Zero-based index at which to start copying the elements from.
+  - Negative index counts back from the end of the array — if `start < 0`, `start + array.length` is used.
+  - If `start < -array.length` or `start` is omitted, `0` is used. 
+  - If `start >= array.length`, nothing is copied. 
+
+- `end` (optional): Zero-based index at which to end copying elements from. `copyWithin()` copies up to the end of the array if `end` is not included.
+  -  Negative index counts back from the end of the array — if `end < 0`, `end + array.length` is used.
+  - If `end < -array.length`, `0` is used.
+  - If `end >= array.length` or `end` is omitted, `array.length` is used, causing all elements until the end to be copied.
+  - If `end` is positioned before or at `start` after normalization, nothing is copied.
+
+### Return value
+
+The modified array.
+
+## Example 1
```

Comment:
```suggestion
### Sparse Arrays

If the `.copyWithin()` method is applied to a sparse array that contains empty slots, it will populate them with an `undefined` value:

## Example
```

In between "empty value:" and "## Example", let's include the following example:


```
console.log([1, , 3].copyWithin(2, 1, 2)); 
// Output: [1, undefined, undefined]
```

We can use a \```js for this one. 😄